bundles/framework/coordinatetool/plugin/CoordinatePluginHandler.js

```diff
@@ -83,28 +84,29 @@ class UIHandler extends StateHandler {
 
     setLon (value) {
         this.updateState({
-            displayXy: {
-                lonlat: {
-                    lat: this.state.displayXy?.lonlat?.lat,
-                    lon: value
-                }
-            }
+            lonField: value
         });
         this.updatePopup();
     }
 
     setLat (value) {
         this.updateState({
-            displayXy: {
-                lonlat: {
-                    lat: value,
-                    lon: this.state.displayXy?.lonlat?.lon
-                }
-            }
+            latField: value
         });
         this.updatePopup();
     }
```

Comment:
It is most likely that both lon and lat are changed at the same time so we should optimize for that and combine the setter to a single one. Otherwise on most cases updatePopup() will be called twice when ever the coordinate changes.

Reply:
Or how does this compare to updateLonLat()?

Reply:
Ah, these are the changelisteners when user writes the coordinates on the field -> popup is refreshed to show the new input value? Maybe we should add some documentation comments here or rename to more descriptive names?

Reply:
Renamed to setXInputValue.